File: examples/sdk/python/runtime.py
from mmos import MMOSClient


def main():

    client = MMOSClient(

        base_url="http://localhost:8080/api/v1",

        api_key="YOUR_API_KEY"
    )

    # ----------------------------------------------------
    # Runtime Health
    # ----------------------------------------------------

    health = client.runtime.health()

    print("Runtime Status :", health.status)
    print("Version        :", health.version)
    print("Uptime         :", health.uptime)

    # ----------------------------------------------------
    # Runtime Information
    # ----------------------------------------------------

    info = client.runtime.info()

    print("\nRuntime")

    print("Environment :", info.environment)
    print("Mode        :", info.mode)

    # ----------------------------------------------------
    # Registered Providers
    # ----------------------------------------------------

    providers = client.runtime.providers()

    print("\nProviders")

    for provider in providers:

        print(
            f"- {provider.name} ({provider.status})"
        )

    # ----------------------------------------------------
    # Runtime Statistics
    # ----------------------------------------------------

    stats = client.runtime.stats()

    print("\nStatistics")

    print("Running Workflows :", stats.running_workflows)
    print("Queued Tasks      :", stats.queued_tasks)
    print("Active Workers    :", stats.active_workers)
    print("CPU Usage         :", stats.cpu_usage)
    print("Memory Usage      :", stats.memory_usage)

    # ----------------------------------------------------
    # Runtime Health Check
    # ----------------------------------------------------

    checks = client.runtime.health_checks()

    print("\nHealth Checks")

    for check in checks:

        print(
            f"- {check.component}: {check.status}"
        )

    # ----------------------------------------------------
    # Runtime Configuration
    # ----------------------------------------------------

    config = client.runtime.configuration()

    print("\nRuntime Configuration")

    print("Execution Mode :", config.mode)
    print("Event Bus      :", config.event_bus)
    print("Memory         :", config.memory)
    print("Cache          :", config.cache)

    # ----------------------------------------------------
    # Graceful Shutdown (Optional)
    # ----------------------------------------------------

    client.runtime.shutdown()

    print("\nRuntime shutdown request accepted.")


if __name__ == "__main__":
    main()

File: examples/sdk/typescript/runtime.ts
import { MMOSClient } from "@mmos-ai/sdk";

async function main() {

    const client = new MMOSClient({

        baseUrl: "http://localhost:8080/api/v1",

        apiKey: "YOUR_API_KEY"
    });

    // ----------------------------------------------------
    // Runtime Health
    // ----------------------------------------------------

    const health = await client.runtime.health();

    console.log("Runtime Status :", health.status);
    console.log("Version        :", health.version);
    console.log("Uptime         :", health.uptime);

    // ----------------------------------------------------
    // Runtime Information
    // ----------------------------------------------------

    const info = await client.runtime.info();

    console.log("\nRuntime");

    console.log("Environment :", info.environment);
    console.log("Mode        :", info.mode);

    // ----------------------------------------------------
    // Registered Providers
    // ----------------------------------------------------

    const providers =
        await client.runtime.providers();

    console.log("\nProviders");

    for (const provider of providers) {

        console.log(
            `${provider.name} (${provider.status})`
        );
    }

    // ----------------------------------------------------
    // Runtime Statistics
    // ----------------------------------------------------

    const stats =
        await client.runtime.stats();

    console.log("\nStatistics");

    console.log(
        "Running Workflows:",
        stats.runningWorkflows
    );

    console.log(
        "Queued Tasks:",
        stats.queuedTasks
    );

    console.log(
        "Active Workers:",
        stats.activeWorkers
    );

    console.log(
        "CPU Usage:",
        stats.cpuUsage
    );

    console.log(
        "Memory Usage:",
        stats.memoryUsage
    );

    // ----------------------------------------------------
    // Graceful Shutdown (Optional)
    // ----------------------------------------------------

    await client.runtime.shutdown();

    console.log("\nRuntime shutdown request accepted.");
}

main().catch(console.error);